scripts/lldbinit.py

```diff
@@ -110,27 +115,31 @@ def print_dump(context: str, expr: str) -> None:
 
     # Try to find a type + id from the input args. If not, the id will be passed
     # through directly to C++, as it can be a variable name.
-    id_type = None
+    found_id_type = False
 
-    # Look for <type><id> as a single argument.
-    if m := re.fullmatch("([a-z_]+)(\\d+)", args[1]):
-        if m[1] in id_types:
+    # Look for <irN>.<type><id> as a single argument.
+    if m := re.fullmatch("ir(\\d+)\\.([a-z_]+)(\\d+)", args[1]):
+        if m[2] in id_types:
             if len(args) != 2:
                 print_usage()
                 return
-            id_type = m[1]
-            print_dump(context, f"{id_types[id_type]}({m[2]})")
+            id_type = m[2]
+            print_dump(
+                context, f"{id_types[id_type]}({context}, {m[1]}, {m[3]})"
+            )
+            found_id_type = True
 
-    # Look for <type> <id> as two arguments.
-    if not id_type:
-        if args[1] in id_types:
-            if len(args) != 3:
+    # Look for <type><id> as a single argument.
+    if m := re.fullmatch("([a-z_]+)(\\d+)", args[1]):
+        if m[1] in untagged_id_types:
```

Comment:
Are you dropping support for dumping `inst17` _without_ `ir.` prefixed?

Reply:
That form does not work anymore, unfortunately. At the moment, trying to dump an inst id just crashes. The InstId needs an ir index to add as a tag.

While we could add the current file's ir index to it, at the moment we're printing them as `irN.instM` so I don't see a lot of point in consuming them in a different way.

I want to change that but it's something beyond just getting dumping working again.

Reply:
You seem to be expecting that it's always `irN`, but it's conditional. See https://github.com/carbon-language/carbon-lang/blob/trunk/toolchain/sem_ir/ids.cpp#L21 or the example https://github.com/carbon-language/carbon-lang/blob/trunk/toolchain/check/testdata/basics/raw_sem_ir/multifile.carbon#L60.

Reply:
Did you look at the link?

https://github.com/carbon-language/carbon-lang/blob/trunk/toolchain/check/testdata/basics/raw_sem_ir/multifile.carbon#L60

```
// CHECK:STDOUT:     inst14:          {kind: Namespace, arg0: name_scope0, arg1: inst<none>, type: type(inst(NamespaceType))}
```

Reply:
Oh, I did.. I was looking at the rhs at arg1. I don't know how we get an inst with a none for the CheckIRId like that, hm...

Reply:
It looks like this `+1` is what prevents that InstId from getting tagged on Add as well: https://github.com/carbon-language/carbon-lang/blob/49213b1ca36cc3283476e3d56b364dd4155f175a/toolchain/sem_ir/file.cpp#L39